src/Modal.js

```diff
@@ -177,4 +35,131 @@ const styles = StyleSheet.create({
   },
 });
 
+const Modal = ({
+  backdropStyle,
+  children,
+  contentStyle,
+  isVisible,
+  onBackdropPress,
+  onHide,
+  ...otherProps
+}) => {
+  const animVal = new Animated.Value(0);
+  const { state, setState } = useState({
+    isVisible: isVisible,
+    isMounted: false,
+  });
+  const { height, width } = useWindowDimensions();
+  const backdropAnimatedStyle = {
+    opacity: animVal.interpolate({
+      inputRange: [0, 1],
+      outputRange: [0, MODAL_BACKDROP_OPACITY],
+    }),
+  };
+  const contentAnimatedStyle = {
+    transform: [
+      {
+        translateY: animVal.interpolate({
+          inputRange: [0, 1],
+          outputRange: [height, 0],
+          extrapolate: "clamp",
+        }),
+      },
+    ],
+  };
+
+  useEffect(() => {
+    setState({
+      ...state,
+      isMounted: true,
+    });
+    if (state.isVisible) {
+      setState({ isVisible: true });
+      Animated.timing(animVal, {
+        easing: Easing.inOut(Easing.quad),
+        // Using native driver in the modal makes the content flash
+        useNativeDriver: false,
+        duration: MODAL_ANIM_DURATION,
+        toValue: 1,
+      }).start();
+    }
+
+    return () => {
+      setState({
+        ...state,
+        isMounted: false,
+      });
+    };
+    // eslint-disable-next-line react-app/react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (isVisible) {
+      setState({ isVisible: true });
+      Animated.timing(animVal, {
+        easing: Easing.inOut(Easing.quad),
+        // Using native driver in the modal makes the content flash
+        useNativeDriver: false,
+        duration: MODAL_ANIM_DURATION,
+        toValue: 1,
+      }).start();
+    } else if (!isVisible) {
+      Animated.timing(this.animVal, {
+        easing: Easing.inOut(Easing.quad),
+        // Using native driver in the modal makes the content flash
+        useNativeDriver: false,
+        duration: MODAL_ANIM_DURATION,
+        toValue: 0,
+      }).start(() => {
+        if (state.isMounted) {
+          setState({ isVisible: false }, onHide);
```

Comment:
I don't think `onHide` will be invoked here. `setState` doesn't have a callback parameter as far as I know.